scripts/dwi_gp_estimation_error_analysis.py

```diff
@@ -164,25 +202,34 @@ def main() -> None:
         # max_iter=2e5,
     )
 
-    # Use Scikit-learn cross validation
-    scores = defaultdict(list, {})
-    for n in args.kfold:
-        for i in range(args.repeats):
-            cv_scores = -1.0 * cross_validate(X, y.T, n, np.max(args.kfold) // n, gpr)
-            scores["rmse"] += cv_scores.tolist()
-            scores["repeat"] += [i] * len(cv_scores)
-            scores["n_folds"] += [n] * len(cv_scores)
-            scores["bval"] += [args.bval_shell] * len(cv_scores)
-            scores["snr"] += [snr_str] * len(cv_scores)
-
-        print(f"Finished {n}-fold cross-validation")
-
-    scores_df = pd.DataFrame(scores)
-    scores_df.to_csv(args.error_data_fname, sep="\t", index=None, na_rep="n/a")
-
-    grouped = scores_df.groupby(["n_folds"])
-    print(grouped[["rmse"]].mean())
-    print(grouped[["rmse"]].std())
+    if args.kfold:
+        # Use Scikit-learn cross validation
+        scores = defaultdict(list, {})
+        for n in args.kfold:
+            for i in range(args.repeats):
+                cv_scores = -1.0 * cross_validate(X, y.T, n, gpr)
+                scores["rmse"] += cv_scores.tolist()
+                scores["repeat"] += [i] * len(cv_scores)
+                scores["n_folds"] += [n] * len(cv_scores)
```

Comment:
```suggestion
                scores["n_folds"] += [n] * len(cv_scores)
                scores["bval"] += [args.bval-shell] * len(cv_scores)
```

I think serializing the b-value is relevant. Revisiting Andersson2015, I've gone further down and have seen Fig 8 (so similar to what we had discussed, now adding different b-values). So I'd recover that.